streams/src/main/java/org/apache/kafka/streams/state/internals/FilteredCacheIterator.java

```diff
@@ -24,11 +24,48 @@
 class FilteredCacheIterator implements PeekingKeyValueIterator<Bytes, LRUCacheEntry> {
     private final PeekingKeyValueIterator<Bytes, LRUCacheEntry> cacheIterator;
     private final HasNextCondition hasNextCondition;
+    private final PeekingKeyValueIterator<Bytes, LRUCacheEntry> wrappedIterator;
 
     FilteredCacheIterator(final PeekingKeyValueIterator<Bytes, LRUCacheEntry> cacheIterator,
-                          final HasNextCondition hasNextCondition) {
+                          final HasNextCondition hasNextCondition,
+                          final CacheFunction cacheFunction) {
         this.cacheIterator = cacheIterator;
         this.hasNextCondition = hasNextCondition;
+        this.wrappedIterator = new PeekingKeyValueIterator<Bytes, LRUCacheEntry>() {
```

Comment:
I'd rather not, since the caller otherwise would have to know that both of those iterators need to share the same underlying iterator. I'd rather have this class encapsulate that.